yaksh/views.py

```diff
@@ -954,3 +958,79 @@ def design_questionpaper(request):
         context = {'form': form, 'questionpaper':True}
         return my_render_to_response('yaksh/design_questionpaper.html',
                                      context, context_instance=ci)
+
+@login_required
+def search_teacher(request, course_id):
+    """ search teachers for the course """
+
+    user = request.user
+    ci = RequestContext(request)
+    context = {}
+    course = get_object_or_404(Course, creator=user, pk=course_id)
+    context['course'] = course
+    if not user.is_authenticated() or not is_moderator(user):
```

Comment:
1. This needs to be checked first, right after obtaining the user (that is it should be at line 967)
1. You can now remove ```user.is_authenticated()``` since ```@login_required``` already checks if user is authenticated